components/skus/browser/skus_service.cc

```diff
@@ -0,0 +1,152 @@
+// Copyright (c) 2021 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// you can obtain one at http://mozilla.org/MPL/2.0/.
+
+#include "brave/components/skus/browser/skus_service.h"
+
+#include <utility>
+
+#include "base/json/json_reader.h"
+#include "base/task/post_task.h"
+#include "base/threading/sequenced_task_runner_handle.h"
+#include "brave/components/skus/browser/pref_names.h"
+#include "brave/components/skus/browser/rs/cxx/src/lib.rs.h"
+#include "brave/components/skus/browser/rs/cxx/src/shim.h"
+#include "brave/components/skus/browser/skus_context_impl.h"
+#include "brave/components/skus/browser/skus_utils.h"
+#include "components/prefs/pref_service.h"
+#include "services/network/public/cpp/shared_url_loader_factory.h"
+
+namespace {
+
+void OnRefreshOrder(skus::RefreshOrderCallbackState* callback_state,
```

Comment:
how does this end up with a raw pointer when we started with a unique_ptr? Where did it get released and why?

Reply:
discussed this with @evq and we are working on some changes for a follow-up